File: client/src/types/PendingHabitConfigurationResponse.ts
import type { HabitCategory } from './HabitCategory'
import type { HabitDifficulty } from './HabitDifficulty'
import type { HabitFrequencyType } from './HabitFrequencyType'

export type PendingHabitConfigurationResponse = {
  effectiveFromDate: string
  category: HabitCategory
  frequencyType: HabitFrequencyType
  targetCount: number
  difficulty: HabitDifficulty
}

File: server/HabitTracker.Api/DTOs/PendingHabitConfigurationResponse.cs
using HabitTracker.Api.Domain.Enums;

namespace HabitTracker.Api.DTOs;

public sealed class PendingHabitConfigurationResponse
{
    public DateOnly EffectiveFromDate { get; set; }

    public HabitCategory Category { get; set; }

    public HabitFrequencyType FrequencyType { get; set; }

    public int TargetCount { get; set; }

    public HabitDifficulty Difficulty { get; set; }
}
